apollo-router/src/apollo_telemetry.rs

```diff
@@ -367,17 +375,21 @@ impl SpanExporter for Exporter {
 
 // Taken from TS implementation
 static GRAPHQL_PARSE_FAILURE: &str = "## GraphQLParseFailure\n";
+#[allow(dead_code)]
 static GRAPHQL_VALIDATION_FAILURE: &str = "## GraphQLValidationFailure\n";
 static GRAPHQL_UNKNOWN_OPERATION_NAME: &str = "## GraphQLUnknownOperationName\n";
 
-fn normalize(op: Option<&opentelemetry::Value>, query: &str) -> String {
-    // If we don't have an operation name, we can't do anything useful
-    // with this query. Just return the appropriate error.
-    let op_name: String = match op {
+fn stats_report_key(op: Option<&opentelemetry::Value>, query: &str) -> String {
+    // This represents a logic error in the router, since we should at
+    // least have "-" as the value for the name if no out of band name
+    // was specified with the query.
+    let mut op_name: String = match op {
         Some(v) => v.as_str().into_owned(),
         None => {
-            tracing::warn!("Could not identify operation name: {}", query);
-            return GRAPHQL_UNKNOWN_OPERATION_NAME.to_string();
+            panic!(
```

Comment:
should the agent really panic if the router has invalid behaviour? Here that means the agent will stop reporting but the router still handles traffic, right? Or is there an agent restart somewhere?

Reply:
It's `panic!` if there is a coding error. i.e. We can only end up here if someone has changed the code in the router and broken the instrumentation. The expected outcome is the router exits abnormally.